### PR TITLE
sdk config now accepts optional headers overrides

### PR DESCRIPTION
**Pull request by scastleman-immutable**
# Summary
optional headers overrides can now be passed through multi-rollup config, allowing the passport sample app to provide an sdk version compatible with the passport v2 endpoint changes

# Impact of Change

## Security

# Other things worth calling out

# Before submitting the PR, follow these steps:

- [ ] Prefix your PR title with `feat: `, `fix: `, `chore: `, `docs:`, or `refactor:`.
- [ ] Prefix your PR title with `breaking-change:` if you have introduced breaking changes to this SDK. Breaking changes are modification that necessitates immediate adjustments by this SDK's users to their applications, clients, or integrations to avert disruptions to existing features or functionalities. Ensure you write clear summary explain the change.

### CODE REVIEW

File: packages/passport/sdk-sample-app/src/context/ImmutableProvider.tsx [hunk 1]
```diff
@@ -41,6 +41,7 @@ const getCoreSdkConfig = (environment: EnvironmentNames) => {
 
 const getPassportConfig = (environment: EnvironmentNames): PassportModuleConfiguration => {
   const sharedConfigurationValues = {
+    headers: { 'x-sdk-version': 'ts-immutable-sdk-1.0.0' },
```
Comment (carmen0208) [inline packages/passport/sdk-sample-app/src/context/ImmutableProvider.tsx:44]: Hey @scastleman-immutable 
Nik also mentioned the **x-sdk-version** is not overridable.
so I think that's probably not going to work

Comment (carmen0208) [reply packages/passport/sdk-sample-app/src/context/ImmutableProvider.tsx:44]: Here is the test to ensure it not override
https://github.com/immutable/ts-immutable-sdk/blob/main/packages/internal/generated-clients/src/config/config.test.ts#L24

Comment (scastleman-immutable) [reply packages/passport/sdk-sample-app/src/context/ImmutableProvider.tsx:44]: They weren't overridable, but I made changes so this gets passed through. Everything worked in testing on my end, but have a go yourself and see if they work for you too

File: packages/passport/sdk/src/config/config.ts [hunk 2]
```diff
@@ -96,6 +96,7 @@ export class PassportConfiguration {
         }),
         passport: createConfig({
           basePath: overrides.passportMrBasePath,
+          headers: overrides.headers,
```
Comment (carmen0208) [inline packages/passport/sdk/src/config/config.ts:99]: This can be dangerous because it is a potential leak for our SDK and risks all the API calls.

Comment (scastleman-immutable) [reply packages/passport/sdk/src/config/config.ts:99]: It's part of the overrides that already allow for custom base paths, relayer URLs, magic keys and more, all of which are accessible for anyone using the SDK. Additionally, there's nothing stopping an attacker from just forking our code and manually adding whatever headers they want anyway, then using that forked code in their application. I don't see this as introducing any risks beyond what we already have as part of our codebase or architecture.

File: packages/passport/sdk-sample-app/src/context/ImmutableProvider.tsx (hunk 1 above)
Comment (carmen0208) [reply packages/passport/sdk-sample-app/src/context/ImmutableProvider.tsx:44]: @CodeSchwert, can you confirm that the header is overriderable?
I did a test myself and noticed that it is overridden. But as you said to me earlier, We didn't expect **createConfig** [function ](https://github.com/immutable/ts-immutable-sdk/blob/main/packages/internal/generated-clients/src/config/config.ts#L21) overridable, 
so my understanding is even if we change the passport config to allow the config override. it still couldn't override the header. I was confused there.

File: packages/passport/sdk/src/config/config.ts (hunk 2 above)
Comment (yundifu) [inline packages/passport/sdk/src/config/config.ts:99]: yeah, i am not convinced that we should add this to fix a sample app issue. as we are exposing headers to other clients, might have risks but meantime i don't know if the headers will be used by other clients.

**Pull request closed without merging**